jdaviz/configs/default/plugins/model_fitting/model_fitting.py

```diff
@@ -683,15 +659,15 @@ def _fit_model_to_spectrum(self, add_data):
         models_to_fit = self._reinitialize_with_fixed()
 
         # Apply masks from selected spectral subsets
-        self._apply_subset_masks(self._spectrum1d, self.spectral_subset)
+        masked_spectrum = self._apply_subset_masks(self.dataset.selected_obj, self.spectral_subset)
```

Comment:
Do you intend to return a subset masked copy of `self.dataset.selected_obj` as `masked_spectrum`? If so, I think this might not do exactly what you're expecting, since the `_apply_subset_masks` method is still updating or creating a mask attribute on `self.dataset.selected_obj`.

Reply:
`self.dataset.selected_obj` is a cached property, so I _think_ this is safe (but since its cached, I guess maybe that could cause an issue).  Do you think I need to manually deepcopy somewhere to prevent multiple masks from being applied on top of each other?  Can you reproduce a case where that happens?

Reply:
ah, you're right, this does cause a problem.  I'll push an update that makes a deepcopy within `_apply_subset_masks`.